src/m365/spo/commands/file/file-list.ts

```diff
@@ -27,32 +30,48 @@ class SpoFileListCommand extends SpoCommand {
   }
 
   public commandAction(logger: Logger, args: CommandArgs, cb: () => void): void {
+
     if (this.verbose) {
       logger.logToStderr(`Retrieving all files in folder ${args.options.folder} at site ${args.options.webUrl}...`);
     }
 
-    let requestUrl: string = `${args.options.webUrl}/_api/web/GetFolderByServerRelativeUrl('${encodeURIComponent(args.options.folder)}')/Files`;
+    this.getFiles(args.options.folder,args).then((files: FilePropertiesCollection): void => {
+      logger.log(files.value);      
+      cb();
+    }, (err: any): void => this.handleRejectedODataJsonPromise(err, logger, cb));
+  }
+
+  // Gets files from a folder recursively.
+  private getFiles(folderUrl: string, args: CommandArgs,files: FilePropertiesCollection={value:[]}) : Promise<FilePropertiesCollection>{
 
+    // If --recursive option is specified, retrieve both Files and Folder details, otherwise only Files.
+    const expandParameters:string = args.options.recursive ? 'Files,Folders':'Files';
+    let requestUrl = `${args.options.webUrl}/_api/web/GetFolderByServerRelativeUrl('${encodeURIComponent(folderUrl)}')?$expand=${expandParameters}`;
     if (args.options.output !== 'json') {
-      requestUrl += '?$select=UniqueId,Name,ServerRelativeUrl';
+      requestUrl += '&$select=Files/UniqueId,Files/Name,Files/ServerRelativeUrl';
     }
-
-    const requestOptions: any = {
+    const requestOptions:any = {
       url: requestUrl,
       method: 'GET',
       headers: {
         'accept': 'application/json;odata=nometadata'
       },
       responseType: 'json'
     };
-
-    request
-      .get<FilePropertiesCollection>(requestOptions)
-      .then((fileProperties: FilePropertiesCollection): void => {
-        logger.log(fileProperties.value);
-
-        cb();
-      }, (err: any): void => this.handleRejectedODataJsonPromise(err, logger, cb));
+    
+    return request.get<FileFolderCollection>(requestOptions)
+      .then((filesAndFoldersResult: FileFolderCollection) => {
+        filesAndFoldersResult.Files.map((file:FileProperties) => files.value.push(file));
```

Comment:
Since we're not using the returned value, we should use `forEach` instead of `map`